remote-api/src/main/java/org/alfresco/rest/api/Rules.java

```diff
@@ -57,4 +59,14 @@
      * @return {@link Rule} definition
      */
     Rule getRuleById(String folderNodeId, String ruleSetId, String ruleId);
+
+    /**
+     * Create new rules (and potentially a rule set if "_default_" is supplied).
+     *
+     * @param folderNodeId The node id of a folder.
+     * @param ruleSetId The id of a rule set (or "_default_" to use/create the default rule set for the folder).
+     * @param rule The definition of the rule.
+     * @return The newly created rules.
```

Comment:
Does the returned list also include existing rules (if any) or only the rules being created, even if some others still exist ?

Does this API return an error if the folder already links to another rule-set ? (EDIT: I can see "checkForLinkedRules" lower-down in RuleServiceImpl)

Reply:
It only returns the newly created rules. I assume this is consistent with other APIs (although I didn't check tbh) - e.g. creating users you wouldn't want all the users in the system returned.